src/NuGet.Core/NuGet.Packaging/Rules/MisplacedAssemblyRule.cs

```diff
@@ -1,61 +1,56 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.IO;
 using System.Linq;
-using NuGet.Packaging;
+using NuGet.Common;
 
-namespace NuGet.Commands.Rules
+namespace NuGet.Packaging.Rules
 {
     internal class MisplacedAssemblyRule : IPackageRule
     {
-        public IEnumerable<PackageIssue> Validate(PackageBuilder builder)
+        public IEnumerable<PackLogMessage> Validate(PackageArchiveReader builder)
         {
-            foreach (IPackageFile file in builder.Files)
+            foreach (var file in builder.GetFiles())
             {
-                string path = file.Path;
-                string directory = Path.GetDirectoryName(path);
+                var directory = Path.GetDirectoryName(file);
```

Comment:
`Directory.GetParent(path).Name` would be more accurate, but it may not make a difference since lib it at the top level in the structure.

Reply:
Directory.GetParent does actual disk I/O, so would be an expensive call here for doing just a string operation.